pkg/kubectl/cmd/util/factory.go

```diff
@@ -176,7 +177,25 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			if cfg.GroupVersion != nil {
 				cmdApiVersion = *cfg.GroupVersion
 			}
-
+			client, err := clients.ClientForVersion(&unversioned.GroupVersion{Version: "v1"})
```

Comment:
I added `--disable-dynamic-discovery` to turn off this behavior.

I also filed:

https://github.com/kubernetes/kubernetes/issues/19853

Since tests are passing with this change.